check-schema-changes has confusing error when released schema is modified

Was fixing my stale PR  [https://github.com/GoogleContainerTools/skaffold/pull/3346](PR) which earlier failed on `make checks` for some import disorder. Merged latest master code and ran ```make checks``` which again failed on ```hack/check-schema-changes.sh```. I am bit confused how master builds are passing. Tried with fresh cloned code on master branch but failing on the ```hack/check-schema-changes.sh```

Contributing guideline suggest to run ```./hack/new_version.sh``` but I am not making changing schema.

### Expected behavior
```make checks``` should run successfully.
### Actual behavior
```
Running validation scripts...
RUN hack/check-schema-changes.sh
INFO[0000] structural changes in latest config, checking on Github if latest is released... 
INFO[0000] Current Skaffold version: v2beta3            
v2beta3 is released, it should NOT be changed!
ERRO[0000] --------
Structural change detected in a released config: pkg/skaffold/schema/latest/config.go
Please create a new PR first with a new version.
You can use 'hack/new_version.sh' to generate the new config version.
If you are running this locally, make sure you have the origin/master branch up to date!
Admin rights are required to merge this PR!
-------- 
diff --git a/pkg/skaffold/schema/latest/config.go b/pkg/skaffold/schema/latest/config.go
index fd08129f4..a1cdee094 100644
--- a/pkg/skaffold/schema/latest/config.go
+++ b/pkg/skaffold/schema/latest/config.go
@@ -22,7 +22,7 @@ import (
        "github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/util"
 )
 
-// This config version is not yet released, it is SAFE TO MODIFY the structs in this file.
+// !!! WARNING !!! This config version is already released, please DO NOT MODIFY the structs in this file.
 const Version string = "skaffold/v2beta3"
 
 // NewSkaffoldConfig creates a SkaffoldConfig
@@ -317,6 +317,15 @@ type ClusterDetails struct {
        // DockerConfig describes how to mount the local Docker configuration into a pod.
        DockerConfig *DockerConfig `yaml:"dockerConfig,omitempty"`
 
+       // ServiceAccountName describes the Kubernetes service account to use for the pod.
+       // Defaults to 'default'.
+       ServiceAccountName string `yaml:"serviceAccount,omitempty"`
+
+       // RunAsUser defines the UID to request for running the container.
+       // If omitted, no SeurityContext will be specified for the pod and will therefore be inherited
+       // from the service account.
+       RunAsUser *int64 `yaml:"runAsUser,omitempty"`
+
        // Resources define the resource requirements for the kaniko pod.
        Resources *ResourceRequirements `yaml:"resources,omitempty"`
```
Output of ./hack/new_version.sh
```
ok      github.com/GoogleContainerTools/skaffold/cmd/skaffold/app       0.048s
FAIL    github.com/GoogleContainerTools/skaffold/cmd/skaffold/app/cmd   0.089s
ok      github.com/GoogleContainerTools/skaffold/cmd/skaffold/app/cmd/config    (cached)
ok      github.com/GoogleContainerTools/skaffold/cmd/skaffold/app/cmd/schema    0.028s
ok      github.com/GoogleContainerTools/skaffold/cmd/skaffold/app/flags (cached)
FAIL    github.com/GoogleContainerTools/skaffold/hack/man       0.059s
ok      github.com/GoogleContainerTools/skaffold/hack/schemas   0.892s
ok      github.com/GoogleContainerTools/skaffold/hack/versions/pkg/diff 0.034s
ok      github.com/GoogleContainerTools/skaffold/hack/versions/pkg/schema       (cached)
ok      github.com/GoogleContainerTools/skaffold/pkg/webhook/labels     (cached)

=== Failed Tests ===
/pkg/skaffold/schema/TestParseExamples
--- FAIL: TestParseExamples (0.00s)
/pkg/skaffold/schema/TestParseExamples/bazel
    TestParseExamples/bazel: samples_test.go:79: unexpected error: skaffold/v2bet3 is an invalid api version
    --- FAIL: TestParseExamples/bazel (0.00s)
panic: interface conversion: util.VersionedConfig is nil, not *latest.SkaffoldConfig [recovered]
        panic: interface conversion: util.VersionedConfig is nil, not *latest.SkaffoldConfig
```

### Information

- Skaffold version: NA
- Operating system: linux
- Contents of skaffold.yaml:
- Go Version: go version go1.13.6 linux/amd64

--- a/hack/check-schema-changes.sh
+++ b/hack/check-schema-changes.sh
@@ -24,8 +24,4 @@
 # If the change is latest and it is released, we fail the PR for the same reason.
 # If the change is in latest and it is not released yet, it is fine to make changes.
 
-
-
-set +x
-
 go run hack/versions/cmd/schema_check/check.go

--- a/hack/versions/pkg/schema/check.go
+++ b/hack/versions/pkg/schema/check.go
@@ -20,14 +20,12 @@ import (
 	"errors"
 	"fmt"
 	"io/ioutil"
-	"os"
 	"path"
 	"strings"
 
 	"github.com/sirupsen/logrus"
 
 	"github.com/GoogleContainerTools/skaffold/hack/versions/pkg/diff"
-	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/color"
 )
 
 const baseRef = "origin/master"
@@ -90,39 +88,37 @@ func RunSchemaCheckOnChangedFiles() error {
 			continue
 		}
 
-		logrus.Infof("structural changes in latest config, checking on Github if latest is released...")
+		logrus.Warnf("Detected changes to the latest config. Checking on Github if it's released...")
 		latestVersion, isReleased := GetLatestVersion()
 		if !isReleased {
-			color.Green.Fprintf(os.Stdout, "%s is unreleased, it is safe to change it.\n", latestVersion)
+			logrus.Infof("Schema %q is not yet released. Changes are ok.", latestVersion)
 			continue
 		}
-		color.Red.Fprintf(os.Stdout, "%s is released, it should NOT be changed!\n", latestVersion)
+
+		logrus.Errorf("Schema %q is already released. Changing it is not allowed.", latestVersion)
+
+		fmt.Printf("\nWhat should I do?\n-----------------\n")
+		fmt.Printf(" + If this retroactive change is required and is harmless to users, indicate on your PR.\n")
+		fmt.Printf(" + Check if a new unreleased version has been created:\n")
+		fmt.Printf("     - Ensure that your branch is up-to-date with the %q branch.\n", baseRef)
+		fmt.Printf("     - Check for a pending PR to create a new version.\n")
+		fmt.Printf(" + Create a separate PR with just the result of running the 'hack/new_version.sh' script.\n")
+
 		filesInError = append(filesInError, configFile)
 	}
 
-	for _, file := range filesInError {
-		logrus.Errorf(changeDetected(file))
-		changes, err := git.diffWithBaseline(file)
-		if err != nil {
-			logrus.Errorf("failed to get diff: %s", err)
-		}
-		fmt.Print(string(changes))
-	}
-
 	if len(filesInError) > 0 {
+		fmt.Printf("\nInvalid changes:\n----------------\n")
+		for _, file := range filesInError {
+			changes, err := git.diffWithBaseline(file)
+			if err != nil {
+				logrus.Errorf("failed to get diff: %s", err)
+			}
+			fmt.Print(string(changes))
+		}
+
 		return errors.New("structural changes detected")
 	}
 
 	return nil
-}
-
-func changeDetected(configFile string) string {
-	return fmt.Sprintf(`--------
-Structural change detected in a released config: %s
-Please create a new PR first with a new version.
-You can use 'hack/new_version.sh' to generate the new config version.
-If you are running this locally, make sure you have the %s branch up to date!
-Admin rights are required to merge this PR!
---------
-`, configFile, baseRef)
 }

--- a/hack/versions/pkg/schema/version.go
+++ b/hack/versions/pkg/schema/version.go
@@ -31,7 +31,7 @@ import (
 
 func GetLatestVersion() (string, bool) {
 	current := strings.TrimPrefix(latest.Version, "skaffold/")
-	logrus.Infof("Current Skaffold version: %s", current)
+	logrus.Debugf("Current Skaffold version: %s", current)
 
 	config, err := ioutil.ReadFile("pkg/skaffold/schema/latest/config.go")
 	if err != nil {

